src/hooks/useDialogArea.tsx

```diff
@@ -10,13 +10,11 @@ export const DialogAreaProvider = ({ ...props }) => (
   <DialogAreaContext.Provider value={useDialogAreaContext()} {...props} />
 );
 
-export const useDialogArea = () => useContext(DialogAreaContext)!;
+export const useDialogArea = () => useContext(DialogAreaContext);
 
 const useDialogAreaContext = () => {
-  const [dialogArea, setDialogArea] = useState<HTMLElement>();
-
+  const dialogAreaRef = useRef(null);
   return {
-    dialogArea,
-    setDialogArea,
+    dialogAreaRef,
   };
```

Comment:
no need to return in object.

`return dialogAreaRef`

Reply:
was thinking about this, but i opted for returning in object for:
1. consistency with hooks
2. extendability. if we ever want to add functionality to this we'll have less code changes if we keep the return in obj format

Reply:
Nice this makes sense